plugins/uefi/fu-self-test.c

```diff
@@ -36,57 +36,6 @@ fu_uefi_pcrs_1_2_func (void)
 	g_assert_cmpint (pcrXs->len, ==, 0);
 }
 
-static void
-fu_uefi_pcrs_2_0_func (void)
-{
-	gboolean ret;
-	g_autoptr(FuUefiPcrs) pcrs = fu_uefi_pcrs_new ();
-	g_autoptr(GError) error = NULL;
-	g_autoptr(GPtrArray) pcr0s = NULL;
-	g_autoptr(GPtrArray) pcr1s = NULL;
-	g_autoptr(GPtrArray) pcrXs = NULL;
-
-	g_setenv ("FWUPD_UEFI_TPM2_YAML_DATA",
-		  "sha1 :\n"
-		  "  0  : cbd9e4112727bc75761001abcb2dddd87a66caf5\n"
-		  "sha256 :\n"
-		  /* old output format of tpm2_listpcrs and tpm2_pcrlist */
-		  "  0  : 122de8b579cce17b0703ca9f9716d6f99125af9569e7303f51ea7f85d317f01e\n"
-		  /* new output format of tpm2_pcrread */
-		  "  1 : 0x0D89E7CA2C487EED36DB5684826B4ABF0904D3BDCD74E61999573570CF3F9C75\n", TRUE);
-
-	ret = fu_uefi_pcrs_setup (pcrs, &error);
-	g_assert_no_error (error);
-	g_assert_true (ret);
-	pcr0s = fu_uefi_pcrs_get_checksums (pcrs, 0);
-	g_assert_nonnull (pcr0s);
-	g_assert_cmpint (pcr0s->len, ==, 2);
-	pcr1s = fu_uefi_pcrs_get_checksums (pcrs, 1);
-	g_assert_nonnull (pcr1s);
-	g_assert_cmpint (pcr1s->len, ==, 1);
-	pcrXs = fu_uefi_pcrs_get_checksums (pcrs, 999);
-	g_assert_nonnull (pcrXs);
-	g_assert_cmpint (pcrXs->len, ==, 0);
-}
```

Comment:
Do you guys have any instrumentation methods to use with the tss stack?  It would be good to keep a test case here, even if it's using a simulated TPM or instrumented data.

Reply:
We usually start an instance of the [IBM TPM simulator](https://sourceforge.net/projects/ibmswtpm2/) to run tests against. tpm2-tss has support for this out of the box, so if you start `tpm_server` in the background and have no physical TPM available, the simulator will be used. Hence we could use something like this as a test if a physical or simulated TPM is available:
```
static void
fu_uefi_pcrs_2_0_func (void)
{
	gboolean ret;
	g_autoptr(FuUefiPcrs) pcrs = fu_uefi_pcrs_new ();
	g_autoptr(GError) error = NULL;
	g_autoptr(GPtrArray) pcr0s = NULL;
	g_autoptr(GPtrArray) pcrXs = NULL;

	/* unset FWUPD_SYSFSTPMDIR so that fu_uefi_pcrs_setup_tpm20 instead of
	   fu_uefi_pcrs_setup_tpm12 is tested */
	g_unsetenv ("FWUPD_SYSFSTPMDIR");

	ret = fu_uefi_pcrs_setup (pcrs, &error);
	g_assert_no_error (error);
	g_assert_true (ret);
	pcr0s = fu_uefi_pcrs_get_checksums (pcrs, 0);
	g_assert_nonnull (pcr0s);
	g_assert_cmpint (pcr0s->len, >=, 2);
	pcrXs = fu_uefi_pcrs_get_checksums (pcrs, 999);
	g_assert_nonnull (pcrXs);
	g_assert_cmpint (pcrXs->len, ==, 0);
}
```
However I don't know how to guarantee that the simulator is started: I tried using [`g_spawn_async`](https://developer.gnome.org/glib/stable/glib-Spawning-Processes.html#g-spawn-async), but that is suboptimal since a failed assertion means that the program is terminated and doesn't kill its child process (and there seems to be no native kill function in GLib anyway, you need to use `kill()` directly).

Reply:
Done, I went for the Arch Linux CI, since that saves me from building the TPM simulator from source and added the above test. It is run if physical TPM is found or`TPM_SERVER_RUNNING` is set, otherwise it is skipped.